models/mli/model_skopes_rules.py

```diff
@@ -0,0 +1,288 @@
+        self.params["max_samples"] = np.random.choice(max_samples) 
+        self.params["max_samples_features"] = np.random.choice(max_samples_features) 
+        self.params["max_depth"] = np.random.choice(max_depth) 
+        self.params["max_features"] = np.random.choice(max_features) 
+        self.params["min_samples_split"] = np.random.choice(min_samples_split) 
+        self.params["bootstrap"] = np.random.choice(bootstrap) 
+        self.params["bootstrap_features"] = np.random.choice(bootstrap_features) 
+
+
+    def _create_tmp_folder(self, logger):
+        # Create a temp folder to store files 
+        # Set the default value without context available (required to pass acceptance test)
+        tmp_folder = os.path.join(temporary_files_path, "%s_SKOPE_model_folder" % uuid.uuid4())
+        # Make a real tmp folder when experiment is available
+        if self.context and self.context.experiment_id:
+            tmp_folder = os.path.join(self.context.experiment_tmp_dir, "%s_SKOPE_model_folder" % uuid.uuid4())
+
+        # Now let's try to create that folder
+        try:
+            os.mkdir(tmp_folder)
+        except PermissionError:
+            # This not occur so log a warning
+            loggerwarning(logger, "SKOPE was denied temp folder creation rights")
+            tmp_folder = os.path.join(temporary_files_path, "%s_SKOPE_model_folder" % uuid.uuid4())
+            os.mkdir(tmp_folder)
+        except FileExistsError:
+            # We should never be here since temp dir name is expected to be unique
+            loggerwarning(logger, "SKOPE temp folder already exists")
+            tmp_folder = os.path.join(self.context.experiment_tmp_dir, "%s_SKOPE_model_folder" % uuid.uuid4())
+            os.mkdir(tmp_folder)
+        except:
+            # Revert to temporary file path
+            tmp_folder = os.path.join(temporary_files_path, "%s_SKOPE_model_folder" % uuid.uuid4())
+            os.mkdir(tmp_folder)
+
+        loggerinfo(logger, "SKOPE temp folder {}".format(tmp_folder))
+        return tmp_folder
+
+
+    def fit(self, X, y, sample_weight=None, eval_set=None, sample_weight_eval_set=None, **kwargs):
+        
+        orig_cols = list(X.names)
+        
+        import pandas as pd
+        import numpy as np
+        from skrules import SkopeRules
+        from sklearn.preprocessing import OneHotEncoder
+        from collections import Counter
+
+              
+        # Get the logger if it exists
+        logger = None
+        if self.context and self.context.experiment_id:
+            logger = make_experiment_logger(experiment_id=self.context.experiment_id,
+                                                 tmp_dir=self.context.tmp_dir,
+                                                 experiment_tmp_dir=self.context.experiment_tmp_dir)
+            
+        # Set up temp folder
+        tmp_folder = self._create_tmp_folder(logger)        
```

Comment:
I don't see a use for this. Do we save anything?

Reply:
Yes, it's saving the skope rules to a csv file.